pkg/provisioners/heartbeats/cmd/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type Heartbeat struct {
+	Sequence int    `json:"id"`
+	Label    string `json:"label"`
+}
+
+var (
+	remote   string
+	label    string
+	period   int
+	sequence int
+	hb       *Heartbeat
+)
+
+func init() {
+	flag.StringVar(&remote, "remote", "", "the host url to heartbeat to")
+	flag.StringVar(&label, "label", "", "a special label")
+	flag.IntVar(&period, "period", 5, "the number of seconds between heartbeats")
+}
+
+func main() {
+	flag.Parse()
+
+	hb = &Heartbeat{
+		Sequence: 0,
+		Label:    label,
+	}
+
+	for {
+		send()
+		time.Sleep(time.Duration(period) * time.Second)
+	}
+}
+
+func send() {
+	hb.Sequence++
+	resp, err := http.Post(remote, "application/json", body())
```

Comment:
here, the source invokes the _target_ (aka `ksvc`), why would it directly invoke HTTP?

I would think think it uses a `channel`, that might have an http implementation (or something else)?

Reply:
I get the `remote` is somehow injected - fine.

I was more asking.  Do we need (perhaps in the future) some higher level API ?  that this would use a `channel` instead of an http client (e.g. when the spec supports other protocols as well).  Perhaps out of scope for now...

Reply:
This is assuming that the channel speaks http. So the source makes a channel, waits for it to resolve, and then gives the pod the resolved channel host as `remote`.

Reply:
It is using a channel. The heartbeat code just does not know it. :D